pageserver/src/tenant.rs

```diff
@@ -1717,7 +1718,33 @@ impl Tenant {
         tenant_id: TenantId,
         remote_storage: Option<GenericRemoteStorage>,
     ) -> Tenant {
-        let (state, _) = watch::channel(state);
+        let (state, mut rx) = watch::channel(state);
+
+        tokio::spawn(async move {
```

Comment:
NIT: Consider using task mgr's version of spawn. Also the fact that we dont store join handle is a little bit concerning but should be fine.

Reply:
Was thinking the same, but then again, using task_mgr here doesn't actually add anything except LOC, since there's only one connection point, which will naturally take care of itself via drop (the watch channel).

Reply:
No, it adds something valuable, namely, that we know that after `task_mgr::shutdown(None, Some(tenant_id), None)`, this task will be gone.

Which is necessary to avoid the following race condition, not covered by this PR:

```
T1: tenant state is changed to Stopping
T1: uninstalls metrics via remove_tenant_metrics
This task: creates new metrics again due to `with_label_values`
T1: drops tenant
This task: exits
```

Result is that now, even though the tenant is gone, there's a stale metric present.